hefquin-engine/src/main/java/se/liu/ida/hefquin/engine/federation/access/impl/response/CardinalityResponseImplWithoutCardinality.java

```diff
@@ -3,10 +3,48 @@
 import se.liu.ida.hefquin.engine.federation.access.DataRetrievalRequest;
 import se.liu.ida.hefquin.engine.federation.access.DataRetrievalResponse;
 
+/**
+ * A subclass of {@link CardinalityResponseImpl} representing a response where the cardinality is not available due to a
+ * possible error.
+ *
+ * This implementation wraps a {@link DataRetrievalResponse} and a {@link DataRetrievalRequest} and captures the
+ * exception that occurred, while assigning {@code Integer.MAX_VALUE} as a placeholder for the unknown cardinality.
+ */
 public class CardinalityResponseImplWithoutCardinality extends CardinalityResponseImpl
 {
+	protected Exception exception;
+
+	/**
+	 * Constructs a {@code CardinalityResponseImplWithoutCardinality} with no specific exception.
```

Comment:
Remove this constructor after implementing [your proposal](https://github.com/LiUSemWeb/HeFQUIN/issues/419#issuecomment-2812048730) for `TPFResponse.getCardinalityEstimate() == null`

Reply:
Now implemented as suggested in [#419](https://github.com/LiUSemWeb/HeFQUIN/issues/419#issuecomment-2821542793)